docs/csharp/pattern-matching.md

```diff
@@ -2,31 +2,203 @@
 title: Pattern Matching | C# Guide
 description: Learn about pattern matching expressions in C#
 keywords: .NET, .NET Core, C#
-ms.date: 10/04/2016
+ms.date: 12/09/2016
+ms.author: wiwagn
 ms.topic: article
 ms.prod: .net
 ms.technology: devlang-csharp
 ms.devlang: csharp
 ms.assetid: 1e575c32-2e2b-4425-9dca-7d118f3ed15b
 ---
 
-# 🔧 Pattern Matching
+# Pattern Matching #
 
-> **Note**
-> 
-> This topic hasn’t been written yet! 
->
-> We welcome your input to help shape the scope and approach. You can track the status and provide input on this
-> [issue](https://github.com/dotnet/docs/issues/1114) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
->
+Pattern matching provides a control flow mechanism that is much more
+generalized than is possible with traditional inheritance hierarchies
+available in Object Oriented practices.
```

Comment:
I don't completely get the contrast between pattern matching and inheritance -- and then the example uses inheritance, which somehow makes it more confusing. Personally, I would relate this not to inheritance, but to convenience and flexibility -- `is` allows assignment and comparison in a single statement, and `switch` offers a flexibility that C# programmers have wanted for a long time.